envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -88,7 +89,10 @@ class EnvoySnapshotFactory(
                 .associateWith { ClusterConfiguration(serviceName = it, http2Enabled = false) }
         }
 
+        // Clusters need to be sorted because if clusters are in different order to previous snapshot then CDS version
```

Comment:
It looks like something that should belong to the contract in java-control-plane's Snapshot, right?